### PR TITLE
Update scaffolding.md

### LINKED ISSUE #3939: Reverse Engineering Page is Unclear On Utilizing User Secrets
The page is very unclear as to how one should actually store user secrets in a way so that
dotnet ef dbcontext scaffold Name=ConnectionStrings:Chinook Microsoft.EntityFrameworkCore.SqlServer
could access them.  Does the command automatically know where to look for the secrets.  Is there a specific format in the secrets file to indicate a field is the database password that should be used?  The page doesn't say.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e1601c3e-3f29-5d84-6350-2b38eeaf6adc
* Version Independent ID: 92a856d6-c5da-e215-46ac-d67c411f3ff9
* Content: [Reverse Engineering - EF Core](https://docs.microsoft.com/en-us/ef/core/managing-schemas/scaffolding?tabs=dotnet-core-cli)
* Content Source: [entity-framework/core/managing-schemas/scaffolding.md](https://github.com/dotnet/EntityFramework.Docs/blob/main/entity-framework/core/managing-schemas/scaffolding.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @bricelam
* Microsoft Alias: **avickers**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #3939

### CODE REVIEW

File: entity-framework/core/managing-schemas/scaffolding.md [hunk 1]
```diff
@@ -37,11 +37,12 @@ Scaffold-DbContext 'Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=Chinook'
 
 ### Configuration and User Secrets
 
-If you have an ASP.NET Core project, you can use the `Name=<connection-string>` syntax to read the connection string from configuration.
+If you have a .NET or ASP.NET Core project, you can use the `Name=<connection-string>` syntax to read the connection string from configuration.
```
Comment (ajcvickers) [inline entity-framework/core/managing-schemas/scaffolding.md:40]: I'm not sure this is helpful. This syntax will only work with .NET applications that are using D.I. and configuration. Usually, this means only ASP.NET Core projects.

Comment (Rick-Anderson) [reply entity-framework/core/managing-schemas/scaffolding.md:40]: > I'm not sure this is helpful. This syntax will only work with .NET applications that are using D.I. and configuration. Usually, this means only ASP.NET Core projects.

The configuration system is .NET, not just ASP.NET Core. See [this comment with a snippet](https://github.com/dotnet/EntityFramework.Docs/issues/3939#issuecomment-1191978026)

Comment (ajcvickers) [reply entity-framework/core/managing-schemas/scaffolding.md:40]: Understood, but how many people use it outside of ASP.NET Core? Which other templates use it by default?

Comment (Rick-Anderson) [reply entity-framework/core/managing-schemas/scaffolding.md:40]: > Understood, but how many people use it outside of ASP.NET Core? Which other templates use it by default?

Plenty of folks who are reverse engineering the DB once they know it works with .NET. As written, the implication is it doesn't work outside of ASP.NET Core and you can't use user secrets with .NET console apps, which is not correct.

Comment (Rick-Anderson) [reply entity-framework/core/managing-schemas/scaffolding.md:40]: See [Dependency injection in .NET](https://docs.microsoft.com/en-us/dotnet/core/extensions/dependency-injection)

It's not just ASP.NET Core.

Comment (ajcvickers) [reply entity-framework/core/managing-schemas/scaffolding.md:40]: How about, "If you have a .NET application that uses the hosting model and configuration system, such as an ASP.NET Core project, then you can use the `Name=<connection-string>` syntax to read the connection string from configuration."

Comment (ErikEJ) [reply entity-framework/core/managing-schemas/scaffolding.md:40]: such as an ASP.NET Core or Worker Service project

Comment (Rick-Anderson) [reply entity-framework/core/managing-schemas/scaffolding.md:40]: @ericstj @IEvangelist please review wording for using configuration with .NET

Comment (ericstj) [reply entity-framework/core/managing-schemas/scaffolding.md:40]: I think @ajcvickers wording sounds OK.  Would it help to provide a link to hosting model and configuration system?  cc @dotnet/area-extensions-hosting @dotnet/area-extensions-configuration

Comment (Rick-Anderson) [inline entity-framework/core/managing-schemas/scaffolding.md:40]:
```suggestion
If you have a .NET application that uses the hosting model and configuration system, such as an ASP.NET Core project, then you can use the `Name=<connection-string>` syntax to read the connection string from configuration.
```

**Pull request merged**